test/unit/dao/QuadraticTimelock.test.ts

```diff
@@ -0,0 +1,335 @@
+import { expectEvent, expectRevert, getAddresses, getImpersonatedSigner, time } from '../../helpers';
+import { expect } from 'chai';
+import hre, { ethers } from 'hardhat';
+import { Signer } from 'ethers';
+
+const toBN = ethers.BigNumber.from;
+
+describe('QuadraticTimelockedDelegator', function () {
+  let userAddress;
+  let secondUserAddress;
+  let beneficiaryAddress1;
+
+  const impersonatedSigners: { [key: string]: Signer } = {};
+
+  before(async () => {
+    const addresses = await getAddresses();
+
+    // add any addresses you want to impersonate here
+    const impersonatedAddresses = [addresses.userAddress, addresses.secondUserAddress, addresses.beneficiaryAddress1];
+
+    for (const address of impersonatedAddresses) {
+      impersonatedSigners[address] = await getImpersonatedSigner(address);
+    }
+  });
+
+  beforeEach(async function () {
+    ({ userAddress, secondUserAddress, beneficiaryAddress1 } = await getAddresses());
+
+    this.tribe = await (await ethers.getContractFactory('MockTribe')).deploy();
+
+    this.window = toBN(4 * 365 * 24 * 60 * 60);
+    this.delegator = await (
```

Comment:
Remove all references to `this` and instead declare variables and their types globally to this test file.

Reply:
+1